examples/terraform-azure-aks-example/main.tf

```diff
@@ -0,0 +1,55 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_resource_group" "k8s" {
+    name     = var.resource_group_name
+    location = var.location
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY AN AZURE KUBERNETES CLUSTER
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_kubernetes_cluster" "k8s" {
+    name                = var.cluster_name
+    location            = azurerm_resource_group.k8s.location
+    resource_group_name = azurerm_resource_group.k8s.name
+    dns_prefix          = var.dns_prefix
+
+    linux_profile {
+        admin_username = "ubuntu"
+
+        ssh_key {
+            key_data = file(var.ssh_public_key)
+        }
+    }
+
+    default_node_pool {
+        name            = "agentpool"
+        node_count      = var.agent_count
+        vm_size         = "Standard_DS1_v2"
+    }
+
+    service_principal {
+        client_id     = var.client_id
+        client_secret = var.client_secret
+    }
+
+    tags = {
+        Environment = "Development"
+    }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE KUBECONFIG FILE
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "local_file" "kubeconfig" {
+    content     = azurerm_kubernetes_cluster.k8s.kube_config_raw
+    filename = "kubeconfig"
```

Comment:
We like to keep our code formatted with `terraform fmt`

Reply:
Done